frontend/src/components/ControlPanel/ControlPanel.tsx

```diff
@@ -29,12 +29,21 @@ function ControlPanel({
 	openDocumentViewer: () => void;
 	addInfoMessage: (message: string) => void;
 }) {
-	const configurableDefences = defences.filter(
-		(defence) => !MODEL_DEFENCES.some((id) => id === defence.id)
+	const configurableDefences =
+		currentLevel === LEVEL_NAMES.SANDBOX
+			? defences
+			: currentLevel === LEVEL_NAMES.LEVEL_3
+			? defences.filter(
+					(defence) => !DEFENCES_HIDDEN_LEVEL3_IDS.includes(defence.id)
+			  )
+			: [];
+
+	const nonModelDefences = configurableDefences.filter(
+		(defence) => !MODEL_DEFENCES.includes(defence.id)
 	);
 
-	const modelDefences = defences.filter((defence) =>
-		MODEL_DEFENCES.some((id) => id === defence.id)
+	const modelDefences = configurableDefences.filter((defence) =>
+		MODEL_DEFENCES.includes(defence.id)
```

Comment:
Filtering / iterating twice seems unnecessary; I came up with a reduce to split it over one iteration but that felt really clunky. This is one area where javascript feels a bit weak... Small beans, leave it!